src/helpers/StatsClient.ts

```diff
@@ -0,0 +1,143 @@
+import { StatsD, Tags } from 'hot-shots';
+import winston = require('winston');
+
+export enum PluginType {
+	AUDIENCE_FEED = 'audience_feed',
+	ACTIVITY_ANALYSER = 'activity_analyser',
+	CUSTOM_ACTION = 'custom_action',
+	AD_RENDERER = 'ad_renderer',
+}
+
+export enum MetricsType {
+	GAUGE = 'gauge',
+	INCREMENT = 'increment',
+}
+
+export interface InitOptions {
+	/**
+	 * The type of your plugin
+	 */
+	pluginType: PluginType;
+	/**
+	 * The id of you plugin depending of the type (feed_id, channel_id, custom_action_id, or ad_renderer_id)
+	 */
+	id: string;
+	/**
+	 * interval to send stats to datadog in ms (default = 10 minutes)
+	 */
+	timerInMs?: number;
+
+	/**
+	 * An optional logger to send Metrics into logs (in debug mode)
+	 */
+	logger?: winston.Logger;
+}
+
+export interface addOrUpdateMetricsOptions {
+	/**
+	 * @example
+	 * ```
+	 * declare your metrics, their types, value and optionals tags.
+	 * {metrics: {processed_users: {type: MetricsType.GAUGE, value: 4, tags: {datamart_id: '4521'}}, users_with_mobile_id_count: {type: MetricsType.INCREMENT, value: 1, tags: {datamart_id: '4521'}}}}
+	 * {processed_users: 4}
+	 */
+	metrics: {
+		[metricName: string]: MetricOptions;
+	};
+}
+
+export type MetricsSet = Map<string, MetricOptions>;
+
+export interface MetricOptions {
+	type: MetricsType;
+	value: number;
+	tags?: Tags;
+}
+
+/**
+ * Send stats to datadog
+ */
+export class StatsClient {
+	private static instance: StatsClient;
+	private interval: NodeJS.Timer;
+	private metrics: MetricsSet;
+	private client: StatsD;
+	private logger?: winston.Logger;
+
+	private constructor(pluginId: { [id: string]: string }, timerInMs: number, logger?: winston.Logger) {
+		this.metrics = new Map();
+		this.logger = logger;
+		this.client = new StatsD({
+			protocol: 'uds',
+			globalTags: { ...pluginId },
+		});
+
+		if (!this.interval) {
+			this.interval = setInterval(() => this.sendStats(), timerInMs);
+		}
+	}
+
+	/**
+	 * @example
+	 * ```
+	 * private this.statsClient: StatsClient
+	 * constructor() {
+	 *   this.statsClient = StatsClient.init({pluginType: PluginType.AUDIENCE_FEED, id: '1234'});
+	 * }
+	 * ```
+	 */
+	static init({ pluginType, id, timerInMs = 10 * 60 * 1000, logger }: InitOptions): StatsClient {
+		const pluginId = this.setPluginId(pluginType, id);
+		return this.instance || (this.instance = new StatsClient(pluginId, timerInMs, logger));
+	}
+
+	private static setPluginId(pluginType: PluginType, id: string): { [id: string]: string } {
```

Comment:
Is it the pluginId you want to set or specifics ids for the given plugin.
My point is that the activity_analyzer_id is not the same as the actual channel id

Reply:
specifics ids for the given plugin. Regarding Simon's metrics list, I though channel id was the feed id would be for an audience feed.

Reply:
plugin_id will be added automatically when the statsD agent is added on the R et D side. 
In my case I guess what I want to add is the plugin_main_id. I can rename the method if you think it is confusing.

Reply:
Yes please :)